execution.py

```diff
@@ -425,7 +435,7 @@ def validate_inputs(prompt, item, validated):
             o_id = val[0]
             o_class_type = prompt[o_id]['class_type']
             r = nodes.NODE_CLASS_MAPPINGS[o_class_type].RETURN_TYPES
-            if r[val[1]] != type_input:
+            if type_input != "*" and r[val[1]] != "*" and r[val[1]] != type_input:
```

Comment:
This change allows for the use of variant types ("*"). It was already supported by the front-end and this was the only change needed to support it on the back-end.

Being able to use variant types let me significantly reduce the number of nodes necessary (by a factor of 6). Specifically, it allows me to create a "Switch" node that works as a switch for every type rather than having a separate "Switch" node for Strings and Ints and Floats and Images etc.

On the other hand, if people use variant types in places where the output type isn't obvious, users might get obscure errors due to hooking up nodes to the wrong types. I could see keeping variant types disabled to prevent overuse (as much as I don't want to make a bunch of duplicate nodes for different types).